react/finger-exercises/redux/02 - react-redux/src/app/screens/App/components/BooksList/index.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { func, arrayOf } from 'prop-types';
+import { bookSelectedPropType } from '@constants/propTypes';
+
+import actionsCreators from '../../../../../redux/book/actions';
+
+import styles from '././styles.scss';
+import Book from './../Book';
+import ShoppingCart from './../ShoppingCart';
+
+class BooksList extends Component {
+  state = {
+    bookSelected: []
+  };
+
+  componentDidMount() {
+    this.props.getBooks();
+  }
+
+  renderBooks = item => <Book key={item.id} data={item} />;
+
+  render() {
+    return (
+      <div>
+        {this.props.books.length ? (
+          this.props.books.map(this.renderBooks)
+        ) : (
+          <div className={styles.noData}>
+            <h2 className={styles.title}>No Data</h2>
+          </div>
+        )}
+        {this.props.bookSelected.length ? <ShoppingCart /> : null}
```

Comment:
If the second argument will be null, you can just do `condition && ...`.

Also, you will have to change this to `this.props.bookSelected.length === 0`, if you don't do that, if length is 0 this will render 0. The number `0` is renderable, as any number, but false, null and undefined aren't

So:

`{this.props.bookSelected.length === 0 && <ShoppingCart />}`